src/main/java/org/apache/camel/idea/completion/extension/YamlPropertyPlaceholdersSmartCompletion.java

```diff
@@ -0,0 +1,130 @@
+                } else if (value instanceof LinkedHashMap) {
+                    buildResultSetForLinkedHashMap(resultSet,virtualFile,keyStr, Collections.singletonList(value));
+                } else {
+                    buildResultSet(resultSet, virtualFile, keyStr,String.valueOf(value));
+                }
+            }
+        });
+    }
+
+    /**
+     * Flat the {@link LinkedHashMap} to string property names and build the {@link CompletionResultSet}
+     */
+    private void buildResultSetForLinkedHashMap(CompletionResultSet resultSet, VirtualFile virtualFile, String keyStr, List<?> propertyList) {
+        propertyList.stream()
+                .filter(l -> (l instanceof LinkedHashMap))
+                .map(LinkedHashMap.class::cast)
+                .flatMap(lhm -> lhm.entrySet().stream())
+                .forEach(e -> {
+                            Map.Entry<String,Object> entry = (Map.Entry)e;
+                            String flatKeyStr = keyStr + "." + String.valueOf(entry.getKey());
+                            if (entry.getValue() instanceof List) {
+                                buildResultSetForList(resultSet, virtualFile, flatKeyStr, (List<?>) entry.getValue());
+                            } else if (entry.getValue() instanceof LinkedHashMap) {
+                                buildResultSetForLinkedHashMap(resultSet,virtualFile,flatKeyStr, Collections.singletonList(entry.getValue()));
+                            } else {
+                                buildResultSet(resultSet, virtualFile, flatKeyStr, String.valueOf(entry.getValue()));
+                            }
+                        }
+
+                );
+    }
+
+    /**
+     * Flat the List to string array and build the {@link CompletionResultSet}
+     */
+    private void buildResultSetForList(CompletionResultSet resultSet, VirtualFile virtualFile, String keyStr, List<?> propertyList) {
+        final AtomicInteger count = new AtomicInteger(0);
+        propertyList.stream()
+                .forEach(e -> {
+                    if (e instanceof String) {
+                        String flatKeyStr = String.format("%s[%s]", keyStr, count.getAndIncrement());
+                        buildResultSet(resultSet, virtualFile, flatKeyStr, String.valueOf(e));
+                    } else if (e instanceof List) {
+                        buildResultSetForList(resultSet,virtualFile,keyStr, (List<?>) e);
+                    } else if (e instanceof LinkedHashMap) {
+                        buildResultSetForLinkedHashMap(resultSet,virtualFile,keyStr,propertyList);
+                    }
+                });
+    }
+
+    private void buildResultSet(CompletionResultSet resultSet, VirtualFile virtualFile, String keyStr,String value) {
+        LookupElementBuilder builder = LookupElementBuilder.create(keyStr + "}}")
+                .appendTailText(value + " :: " + virtualFile.getPresentableName(), true)
```

Comment:
Use `[filename]` syntax for the filename. And maybe try to st it as the java type instead so its right aligned.